stock_picking_accounting_date/models/stock_move.py

```diff
@@ -14,10 +14,20 @@ class StockMove(models.Model):
 
     @api.depends("date", "accounting_date")
     def _compute_accounting_date(self):
-        for line in self:
-            line.accounting_date = fields.Datetime.context_timestamp(self, line.date)
-            if line.picking_id.accounting_date:
-                line.accounting_date = line.picking_id.accounting_date
+        # 'force_period_date' context is assigned when user sets accounting date in
+        # inventory adjustment
+        force_period_date = self._context.get("force_period_date")
+        if force_period_date:
+            for line in self:
+                line.accounting_date = force_period_date
```

Comment:
Can we remove loop and assign as recordset? I think we don't need to use loop here because there is no specific condition for each record.

Reply:
Thanks! Updated.